### ISSUE TITLE
The server actively disconnects, the SDK cannot perceive(SDK v4.0.4beta)

### ISSUE THREAD

**Issue by MrZhuGe** [aws/aws-iot-device-sdk-embedded-C#524]
Hello, I found a problem during the use. I use the SDK connection service on one device, and use the same device information connection service on the other device. The server will actively disconnect the first device, and then the SDK does not sense it and does not give it. Any callback information is blocked on the heartbeat. In previous versions of the SDK, it was able to detect that the server was actively disconnected.

**Comment by MrZhuGe** [comment]
@gordonwang0 I think that the server actively closes the connection of the device, there should be a disconnect callback, right?

**Comment by gordonwang0** [comment]
Hi @MrZhuGe,

You are correct, currently server-to-client disconnects are not reported through the disconnect callback.  We will implement this shortly.

**Comment by gordonwang0** [comment]
By the way -- it is expected for your application to block until the MQTT keep-alive, as that is how MQTT can detect the server-side closed connection.  Any other ways to detect a server-closed connection, while possible, are outside of the MQTT spec.

**Comment by MrZhuGe** [comment]
It is convenient to ask about the handling of this problem. We are using it. This issue has a relatively large impact on us, thank you.

**Comment by abhidixi11** [comment]
I have created internal issue to track this feature request.

**Comment by MrZhuGe** [comment]
@abhidixi11 @gordonwang0 Excuse me, is there a solution to this problem?

**Comment by lundinc2** [comment]
Hello @MrZhuGe,

We currently have not implemented this feature yet. In the meantime, would it be possible for you to use the MQTT keep alive mechanism described earlier by @godonwang0 in your application?

**Comment by gkwicker** [comment]
Hi @MrZhuGe, PR #634 adds a close callback to the network interface, which you should be able to use to detect the server close condition.  I'll leave this issue open for now, but if that works for you please let me know.

**Comment by MrZhuGe** [comment]
Hi @gkwicker, This callback is useful and has been verified, thank you very much.

**Comment by abhidixi11** [comment]
Hello MrZhuGe,
Thanks for your feedback. I'm closing this issue. If you have any further questions, please feel free to open it again, or open a new issue.
Thanks.
